src/middleware/validation.middleware.ts

```diff
@@ -58,11 +58,20 @@ export function checkValidations(req: Request, res: Response, next: NextFunction
       }
       const entityNumber = appData?.[EntityNumberKey];
 
+      let signOutPreviousPagePrefix = "";
+      let signOutExtraQueryParams = "";
+
       // The journey property may already be part of the page form data/body so get it from there and override it if we are on a remove journey
       // Then when we pass it back into the template, make sure it is below/after the req.body fields so it overrides the req.body value
       let journey = req.body["journey"];
-      if (isRemoveJourney(req)){
+      if (isRemoveJourney(req)) {
         journey = config.JourneyType.remove;
+        if (req.originalUrl.includes(`/${config.REMOVE_SECTION}`)) {
+          signOutPreviousPagePrefix = config.REMOVE_SECTION;
+        }
+        // need to remove the journey param if there is one as the sign-out-user-banner will add one if on remove journey
+        const queryParams: string[] = getQueryParamsWithExclusion(req, config.JOURNEY_QUERY_PARAM);
```

Comment:
Could this check be done in the sign-out-user-banner template logic rather than here? i.e. collect as much of the logic there as possible.